### PR TITLE
Role module

### PR DESCRIPTION
**Pull request by Zlet20**

### CODE REVIEW

File: backend/backend-e-learning/src/main/java/com/group6/java16/cybersoft/role/service/RoleServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        return new PageResponseModel<>(rp.getNumber() + 1, rp.getTotalPages(), 
+            rp.getContent().stream().map(RoleMapper.INSTANCE::toResponseDTO).collect(Collectors.toList()));
+    
+		
+	}
+
+
+	@Override
+	public RoleResponseDTO update(String id, @Valid RoleDTO dto) {
+		Optional<ELRole> roleOpt = roleRepository.findById(UUID.fromString(id));
```
Comment (NamNguyen03) [inline backend/backend-e-learning/src/main/java/com/group6/java16/cybersoft/role/service/RoleServiceImpl.java:86]: use service hepler

File: backend/backend-e-learning/src/main/java/com/group6/java16/cybersoft/role/service/RoleServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	public RoleResponseDTO createRole(@Valid RoleDTO dto) {
+		
+		return RoleMapper.INSTANCE.toResponseDTO(roleRepository.save(RoleMapper.INSTANCE.toModel(dto)));
+	}
+
+	
+	@Override
+	public PageResponseModel<RoleResponseDTO> search(PageRequestModel pageRequestModel) {
+		int page = pageRequestModel.getPageCurrent()-1;
+        int size = pageRequestModel.getItemPerPage();
```
Comment (NamNguyen03) [inline backend/backend-e-learning/src/main/java/com/group6/java16/cybersoft/role/service/RoleServiceImpl.java:56]: add space

File: fontend/Admin/src/app/api-clients/role.client.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    deleteById(id:string): Observable<Response<string>>{
+        
+        
+        return this.httpClient.delete<Response<string>>(this.apiEndpoint+"/"+id);
+
+    }
+    updateById(id:string,rq:BaseRole): Observable<Response<RoleResponse>>{
+        
+        
+        return this.httpClient.delete<Response<RoleResponse>>(this.apiEndpoint+"/"+id);
```
Comment (NamNguyen03) [inline fontend/Admin/src/app/api-clients/role.client.ts]: update, why use method delete

File: backend/backend-e-learning/src/main/java/com/group6/java16/cybersoft/role/service/RoleService.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import com.group6.java16.cybersoft.role.dto.RoleDTO;
+import com.group6.java16.cybersoft.role.dto.RoleResponseDTO;
+
+public interface RoleService {
+
+	RoleResponseDTO createRole(@Valid RoleDTO dto);
+
+	PageResponseModel<RoleResponseDTO> search(PageRequestModel pageRequestModel);
+
+	RoleResponseDTO update(String id, @Valid RoleDTO dto);
```
[the remark this answers is not in the document]
Comment (Zlet20) [reply backend/backend-e-learning/src/main/java/com/group6/java16/cybersoft/role/service/RoleService.java:16]: done

**Pull request merged**